function/spring-computer-vision-function/src/test/java/org/springframework/cloud/fn/computer/vision/ComputerVisionFunctionConfigurationTests.java

```diff
@@ -0,0 +1,391 @@
+				assertThat(context).hasBean("imageClassifications");
+
+				Function<Message<byte[]>, Message<byte[]>> predictor = (Function<Message<byte[]>, Message<byte[]>>) context
+					.getBean("imageClassifications", Function.class);
+
+				var djlProperties = context.getBean(DjlConfigurationProperties.class);
+
+				assertThat(djlProperties.getApplicationType()).isEqualTo(ApplicationType.IMAGE_CLASSIFICATION);
+				assertThat(djlProperties.getInputClass()).isEqualTo(Image.class);
+				assertThat(djlProperties.getOutputClass()).isEqualTo(Classifications.class);
+
+				byte[] inputImage = new ClassPathResource("/karakatschan.jpg").getInputStream().readAllBytes();
+
+				Message<byte[]> outputMessage = predictor.apply(MessageBuilder.withPayload(inputImage).build());
+
+				assertThat(outputMessage).isNotNull();
+				assertThat(outputMessage.getPayload()).isNotNull();
+				assertThat(outputMessage.getPayload().length).isGreaterThan(0);
+				assertThat(outputMessage.getHeaders()).containsKey("cvjson");
+
+				String json = outputMessage.getHeaders().get("cvjson", String.class);
+
+				log.info(json);
+
+				assertThat(JsonHelper.toClassifications(json)).isNotNull();
+			});
+	}
+
+	@Test
+	public void poseEstimation() {
+		applicationContextRunner.withPropertyValues(
+		// @formatter:off
+					"computer.vision.function.augment-enabled=true",
+					"djl.application-type=" + ApplicationType.POSE_ESTIMATION,
+					"djl.input-class=" + Image.class.getName(),
+					"djl.output-class=" + Joints.class.getName(),
+					"djl.arguments.threshold=0.3",
+					"djl.model-filter.backbone=resnet18",
+					"djl.model-filter.flavor=v1b",
+					"djl.model-filter.dataset=imagenet")
+				// @formatter:on
+			.run((context) -> {
+				assertThat(context).hasSingleBean(ZooModel.class);
+				assertThat(context).hasBean("predictorProvider");
+				assertThat(context).hasBean("poseEstimation");
+
+				Function<Message<byte[]>, Message<byte[]>> predictor = (Function<Message<byte[]>, Message<byte[]>>) context
+					.getBean("poseEstimation", Function.class);
+
+				var djlProperties = context.getBean(DjlConfigurationProperties.class);
+
+				assertThat(djlProperties.getApplicationType()).isEqualTo(ApplicationType.POSE_ESTIMATION);
+				assertThat(djlProperties.getInputClass()).isEqualTo(Image.class);
+				assertThat(djlProperties.getOutputClass()).isEqualTo(Joints.class);
+
+				byte[] inputImage = new ClassPathResource("/pose.png").getInputStream().readAllBytes();
+
+				Message<byte[]> outputMessage = predictor.apply(MessageBuilder.withPayload(inputImage).build());
+
+				assertThat(outputMessage).isNotNull();
+				assertThat(outputMessage.getPayload()).isNotNull();
+				assertThat(outputMessage.getPayload().length).isGreaterThan(0);
+				assertThat(outputMessage.getHeaders()).containsKey("cvjson");
+
+				String ssJson = outputMessage.getHeaders().get("cvjson", String.class);
+
+				log.info(ssJson);
+
+				save(outputMessage.getPayload(), "pose-estimation.jpg");
```

Comment:
Why do we need this `save()` everywhere if we don't assert against the final file in the `build` ?
And to be honest, it is better to store into some `@TempDir` instead to not interfere with the rest of build artifacts.